source/Halibut/Transport/Protocol/ControlMessageReader.cs

```diff
@@ -57,17 +59,18 @@ internal async Task<string> ReadUntilNonEmptyControlMessageAsync(Stream stream)
 
         internal async Task<string> ReadControlMessageAsync(Stream stream)
         {
+            var cts = new CancellationTokenSource(stream.ReadTimeout);
             StringBuilder sb = new StringBuilder();
             while (true)
             {
                 var nextByte = new byte[1];
-                var read = await stream.ReadAsync(nextByte, 0, nextByte.Length);
+                var read = await stream.ReadAsync(nextByte, 0, nextByte.Length, cts.Token);
```

Comment:
TODO under net48 we can't pass a cts, for now if we do go ahead with this I would just not pass the token in net48. Since we don't have any net48 polling clients.

Reply:
Some how the build thinks this is ok in net48, I was sure stream didn't have an async overload which takes a CT. Looking at the portforwarder it is using sockets rather than streams. I guess it is sockets that doesn't have the overload that takes the CT.